tests/playbooks/tests_infiniband.yml

```diff
@@ -0,0 +1,160 @@
+# SPDX-License-Identifier: BSD-3-Clause
+---
+- hosts: all
+  tasks:
+    - name: "INIT Prepare setup"
+      debug:
+        msg: "##################################################"
+    - block:
+        # For initscripts provider, the script
+        # `/etc/sysconfig/network-scripts/ifup-ib` is required in order to
+        # configure the infiniband profiles
+        - name: Validate the script 'ifup-ib' existed
+          command: test -x /etc/sysconfig/network-scripts/ifup-ib
+          register: ifup_ib_exist
+          failed_when: ifup_ib_exist.rc == 1
+          when: network_provider == "initscripts"
+        - name: "TEST Configuring infiniband devices"
+          debug:
+            msg: "##################################################"
+        - name: Setup the IP over the infiniband device for nm provider
+          import_role:
+            name: linux-system-roles.network
+          vars:
+            network_connections:
+              - name: ib0
+                type: infiniband
+                interface_name: ib0
+
+              - name: ib0-10
+                interface_name: ib0.000a
+                type: infiniband
+                autoconnect: yes
+                infiniband:
+                  p_key: 10
+                  transport_mode: datagram
+                parent: ib0
+                state: up
+                ip:
+                  dhcp4: no
+                  auto6: no
+                  address:
+                    - 198.51.100.133/30
+                    - 2001:db8::2/32
+          when: network_provider == 'nm'
```

Comment:
What is the reason to have different network_connections depending on the provider?

Reply:
I simplied the code to only have one network_connections variable. And you may notice that the ipoib interface name changed and pkey value changed when using different network provider, it is determined by how initscripts and NM backend would activate the connection. 

Reference;
1. https://github.com/linux-system-roles/network/pull/493#issuecomment-1101554399 
2. https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/networking_guide/sec-configuring_ipoib